ddocs/erlang_filters/filters/doc_by_place.erl

```diff
@@ -0,0 +1,98 @@
+% Benchmarking code for Erlang REPL nicked and modified to remove module param from here:
+%  https://erlangcentral.org/wiki/index.php/Measuring_Function_Execution_Time
+% Params: FnName, Params, Rounds, eg:
+%  TC(DeepGet, [DeepGet, [<<"userCtx">>, <<"name">>], Req], 100000).
+% TC = fun(TC_F, TC_A, TC_N) when TC_N > 0 -> TC_L = tl([begin {TC_T, _Result} = timer:tc(TC_F, TC_A), TC_T end || _ <- lists:seq(1, TC_N)]), TC_Min = lists:min(TC_L), TC_Max = lists:max(TC_L), TC_Med = lists:nth(round((TC_N - 1) / 2), lists:sort(TC_L)), TC_Avg = round(lists:foldl(fun(TC_X, TC_Sum) -> TC_X + TC_Sum end, 0, TC_L) / (TC_N - 1)), io:format("Range: ~b - ~b mics~nMedian: ~b mics ~nAverage: ~b mics~n", [TC_Min, TC_Max, TC_Med, TC_Avg]), TC_Med end.
+
+% Made "cmd+v futon-ready" with the following disgusting display of black magic
+% cat scratch.erl | sed 's/%.*//' | sed 's/"/\\"/'g | sed -e ':a' -e 'N' -e '$!ba' -e 's/\n/ /g' | pbcopy
+fun ({Doc}, {Req}) ->
+
+  % Some values are wrapped in a tuple for some arbitrary reason
+  %  i.e. {[{...}]} instead of [{...}]
+  % Instead of working this out we're just guarding against it
+  % TODO work out all the rules for this so we can optimise this function
+  %      away for better performance
+  SafeGetValue = fun(Key, PropListMaybe, Default) ->
+    if
+      is_tuple(PropListMaybe) andalso size(PropListMaybe) == 1 ->
+        proplists:get_value(Key, element(1, PropListMaybe), Default);
+      is_list(PropListMaybe) ->
+        proplists:get_value(Key, PropListMaybe, Default);
+      true ->
+        Default
+    end
+  end,
+
+  DeepGet = fun(DeepGet, KeyList, PropList) ->
```

Comment:
Yep you could probably do that (though trying to run it is a pretty critical step, a lot of this code is trial and error because Erlang is not something I'm familiar with). If you think that helps readability an amount that makes it worth investigating I can check it out (your call, can't tell it's my code). Note that I put quite a bit of work into removing as many function calls as possible because in this state (Couch injecting and evaling code into the erlang engine) function calls seem to be quite slow, and so I do worry this would add another layer and slow it down.

(side note: this entire file is more complicated to read than when I started because I've made some of my clean code a lot uglier for performance reasons. This is why I'm keen to keep the JS impl around, for readability, backed by good test coverage).